Python/ast.c

```diff
@@ -1464,11 +1537,14 @@ ast_for_arguments(struct compiling *c, const node *n)
                 if (!arg)
                     return NULL;
                 asdl_seq_SET(posargs, k++, arg);
-                i += 2; /* the name and the comma */
+                i += 1; /* the name */
+                if (TYPE(CHILD(n, i)) == COMMA)
```

Comment:
I tracked down at least one crash to this line when calling `compile(mod, fn, "exec")` on the `inspect` module in `test_ast.ASTValidatorTests.test_stdlib_validates`.

`NCH(n)` needs to be checked here as it is correct (for what has been allocated - I haven't tried to figure out whether there ought to be another child here). The occasions where this causes a crash is because `n->n_child[0]` is right at the end of the allocated page, and so accessing `n->n_child[1]` triggers an access violation rather than reading into uninitialized heap space.

I can't answer precisely why it only occurs on 32-bit Windows, but it does.

Reply:
I've run it a few times now and it has always crashed at this line, so this may be it. I don't have any more time right now to dig in or review all the other changes like this, I'm sorry.

Reply:
Thanks! This looks indeed very likely -- adding a guard for `i < NCH(n)` here and in a couple of nearby similar spots solved the other mysterious problem I alluded to last night. Once I have some other, unrelated changes ready I'll push again and we'll see how it goes.